analyzers/src/SonarAnalyzer.Common/Rules/DebuggerDisplayUsesExistingMembersBase.cs

```diff
@@ -0,0 +1,89 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Text.RegularExpressions;
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class DebuggerDisplayUsesExistingMembersBase<TAttributeSyntax, TSyntaxKind> : SonarDiagnosticAnalyzer<TSyntaxKind>
+    where TAttributeSyntax : SyntaxNode
+    where TSyntaxKind : struct
+{
+    private const string DiagnosticId = "S4545";
+
+    private readonly Regex noQuotesModifierExpressionRegex = new(@",\s*nq\s*$", RegexOptions.None, RegexConstants.DefaultTimeout);
+    private readonly Regex evaluatedExpressionRegex = new(@"\{(?<EvaluatedExpression>[^\}]+)\}", RegexOptions.Multiline, RegexConstants.DefaultTimeout);
+
+    protected abstract SyntaxNode GetAttributeFormatString(TAttributeSyntax attribute);
+    protected abstract bool IsValidMemberName(string memberName);
+
+    protected override string MessageFormat => "'{0}' doesn't exist in this context.";
+
+    protected DebuggerDisplayUsesExistingMembersBase() : base(DiagnosticId) { }
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterNodeAction(Language.GeneratedCodeRecognizer,
+            c =>
+            {
+                var attribute = (TAttributeSyntax)c.Node;
+                if (Language.Syntax.IsKnownAttribute(attribute, KnownType.System_Diagnostics_DebuggerDisplayAttribute, c.SemanticModel)
+                    && GetAttributeFormatString(attribute) is { } formatString
+                    && FirstInvalidMemberName(c, formatString.GetFirstToken().ValueText, attribute) is { } firstInvalidMember)
+                {
+                    c.ReportIssue(Diagnostic.Create(Rule, formatString.GetLocation(), firstInvalidMember));
+                }
+            },
+            Language.SyntaxKind.Attribute);
+
+    private string FirstInvalidMemberName(SonarSyntaxNodeReportingContext context, string formatString, TAttributeSyntax attributeSyntax)
+    {
+        try
+        {
+            foreach (Match match in evaluatedExpressionRegex.Matches(formatString))
+            {
+                if (match.Groups["EvaluatedExpression"] is { Success: true, Value: var evaluatedExpression }
+                    && ExtractValidMemberName(evaluatedExpression) is { } memberName
+                    && GetAttributeTarget(attributeSyntax) is { } targetSyntax
+                    && context.SemanticModel.GetDeclaredSymbol(targetSyntax) is { } targetSymbol
+                    && GetTypeContainingReferencedMembers(targetSymbol) is { } typeSymbol
+                    && typeSymbol.GetSelfAndBaseTypes().SelectMany(x => x.GetMembers()).All(x => Language.NameComparer.Compare(x.Name, memberName) != 0))
```

Comment:
The all members should be precomputed into a hashset. Here should be just `.Contains(..)`